assets/js/frontend/course-video/video-extension.js

```diff
@@ -0,0 +1,63 @@
+( () => {
+	function handleVisibilityChange( video ) {
+		return function () {
+			if ( document.hidden ) {
+				video.pause();
+			}
+		};
+	}
+
+	function preventClick( event ) {
+		event.preventDefault();
+		return false;
+	}
+
+	function onEnded() {
+		if ( window.sensei.courseVideoSettings.courseVideoRequired ) {
+			document
+				.querySelectorAll( '[data-id="complete-lesson-button"]' )
+				.forEach( ( button ) => {
+					button.removeEventListener( 'click', preventClick );
+					button.disabled = false;
+				} );
+		}
+
+		if ( window.sensei.courseVideoSettings.courseVideoAutoComplete ) {
+			// submit complete lesson form
+			document
+				.querySelectorAll( '[data-id="complete-lesson-form"]' )
+				.forEach( ( form ) => {
+					form.submit();
+				} );
+		}
+	}
+
+	function initPlayer( video ) {
```

Comment:
I think we could re-use the code from all the different players. It seems only the `initPlayer` function would change, right?

Just the event seems to change a little, so we could handle the event in the `initPlayer` in a way where they could call the same `onEnded` function.